tm2/pkg/bft/consensus/state.go

```diff
@@ -1793,6 +1915,44 @@ func (cs *ConsensusState) logTelemetry(block *types.Block) {
 	metrics.BlockSizeBytes.Record(context.Background(), int64(block.Size()))
 }
 
+func (cs *ConsensusState) logErrorWithTrace(span trace.Span, msg string) {
+	if span != nil {
+		span.RecordError(errors.New(msg))
+		span.SetStatus(codes.Error, msg)
+	}
+	cs.Logger.Error(msg)
+}
+
+func (cs *ConsensusState) panicWithTrace(span trace.Span, msg string) {
+	if span != nil {
+		span.RecordError(errors.New(msg))
+		span.SetStatus(codes.Error, msg)
+	}
+	panic(msg)
+}
+
+func (cs *ConsensusState) addTrace(spanName string, opts ...trace.SpanStartOption) trace.Span {
+	var span trace.Span
+
+	if telemetry.TracesEnabled() {
+		var ctx context.Context
+		spanNameWithInfo := fmt.Sprintf("%s - CS(%v/%v/%v)", spanName, cs.Height, cs.Round, cs.Step)
+		spanNameWithInfo = strings.ReplaceAll(spanNameWithInfo, "RoundStep", "RS")
```

Comment:
This seems redundant. Why not specify them once in the root span and keep them as attributes for other spans?
I think we could retain the step attributes in the name, though.

Reply:
Yeah, I think some info like Height is redundant,  maybe I should keep the Round + Step ?

Or maybe better, I can separate round in another trace root so we can keep just step and make it easier to trace ?